clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -274,11 +288,24 @@ public void prepareResponseFrom(AbstractResponse response, Node node, boolean di
      * @param disconnected Whether the request was disconnected
      */
     public void prepareResponse(RequestMatcher matcher, AbstractResponse response, boolean disconnected) {
-        prepareResponseFrom(matcher, response, null, disconnected);
+        prepareResponseFrom(matcher, response, null, disconnected, false);
+    }
+
+    /**
+     * Raise an unsupported version error on the next request. Raise an exception if the next request
+     * does not match the provided matcher function.
```

Comment:
Hmm, does this throw an unsupported version error if the next request matches the matcher and a different exception if it does not? The description could be a bit clearer.